Update reference-target-explainer.md

alice opened the pull request:
- Add clearer use cases.
- Have the phases refer to problems rather than solutions.
- Reflect the lack of consensus on "phase 2".

alice committed:
Update reference-target-explainer.md.

- Add clearer use cases.
- Have the phases refer to problems rather than solutions.
- Reflect the lack of consensus on "phase 2".

alice committed:
Fix typos and confusing wording.

alice commented on proposals/reference-target-explainer.md:
```diff
@@ -318,9 +428,12 @@ Some JavaScript attributes reflect HTML attributes as Element objects rather tha
+> The `.form` and `.list` attributes are currently specced to be `HTMLFormElement` or `HTMLDataListElement`,
+> so they should be updated so the host element can be returned.
+> However, the underlying association still exists even if the IDL getter returns null:
+> the input would be connected to the form, for example; it's just not reflected by the `.form` attribute.
```
This sentence is potentially confusing: I mean to convey that even if the IDL attribute's type isn't compatible, the association is still valid. I'm not sure whether it's better to just remove this sentence, or to re-word it somehow to be less confusing.

dandclark replied:
IMO it's best to just remove it. Based on how https://github.com/WICG/webcomponents/issues/1072 wrapped up it looks like the IDL types will be made compatible so this isn't an issue.

alice replied:
```suggestion
```

Happy to remove it :)

keithamus commented on proposals/reference-target-explainer.md:
```diff
@@ -1,120 +1,332 @@
 # Reference Target for Cross-Root ARIA
 
-Author: [Ben Howell](https://github.com/behowell)
+Original proposal author: [Ben Howell](https://github.com/behowell)
```
```suggestion
Original proposal author: [Ben Howell](https://github.com/behowell)
Current Author: [Alice Boxhall](https://github.com/alice)
```

alice replied:
I decided to leave my name off as I'm not really adding any new explicit API proposals, but at the same time I didn't want to give the impression that Ben was the sole author and thus to blame for all the wording :)

Westbrook replied:
Maybe `Proposal editor`?

dandclark replied:
Or just:
```suggestion
##Authors
- [Ben Howell](https://github.com/behowell)
- [Alice Boxhall](https://github.com/alice)
```

alice replied:
Can I just leave it? 😅

keithamus commented on proposals/reference-target-explainer.md:
```diff
@@ -1,120 +1,332 @@
-Web components have an increasing number of features that allow them to work and act like builtin elements. For example:
+For example, Spectrum Web Components' [`sp-checkbox` component](https://opensource.adobe.com/spectrum-web-components/components/checkbox/)
+composes an `<input type=checkbox>`,
+augmenting its functionality in several ways including adding an `indeterminate` state:
```
It might be worth clarifying this augments by adding a `indeterminate` attribute which (I imagine) reflects the `indeteminate` IDL:


https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/indeterminate

I was a little confused initially reading this as I thought `indeterminate` was part of HTML, but it's just the IDL not the attribute.

alice replied:
Ahh, I see, that's why there's no `aria-checked` attribute.

keithamus commented on proposals/reference-target-explainer.md:
```diff
@@ -1,120 +1,332 @@
-  >
-    <input id="real-input">
+<button commandFor="pets">
+<md-dialog id="pets>
```
```suggestion
<md-dialog id="pets">
```

alice committed:
Fix missing close quotation mark

Co-authored-by: Keith Cirkel <keithamus@users.noreply.github.com>

alice committed:
Update explanation of sp-checkbox

@keithamus pointed out that it uses https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/indeterminate

dandclark reviewed (approved):
Thank you for making these improvements!

alice committed:
Remove confusing, out of date sentence about IDL getters.
